include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_reduce_then_scan.h

```diff
@@ -266,42 +266,89 @@ __scan_through_elements_helper(const __dpl_sycl::__sub_group& __sub_group, _GenI
     }
 }
 
+// The default sub-group size for reduce-then-scan is 32, but we conditionally enable sub-group sizes of 16 on Intel
+// devices to workaround a hardware bug. From the host side, return 32 to assert that this sub-group size is supported
+// by an arbitrary device.
+constexpr inline std::uint8_t
+__get_reduce_then_scan_reqd_sg_sz_host()
+{
+    return 32;
+}
+
+// To workaround a hardware bug on certain Intel iGPUs with older driver versions and -O0 device compilation, use a
+// sub-group size of 16. Note this function may only be called on the device as _ONEDPL_DETECT_SPIRV_COMPILATION is only
+// valid here.
+constexpr inline std::uint8_t
+__get_reduce_then_scan_actual_sg_sz_device()
+{
+    return
+#if _ONEDPL_DETECT_COMPILER_OPTIMIZATIONS_ENABLED || !_ONEDPL_DETECT_SPIRV_COMPILATION
+        32;
+#else
+        16;
+#endif
+}
+
+inline auto
+__get_reduce_then_scan_sub_group_params(std::uint32_t __work_group_size, std::uint8_t __sub_group_size,
+                                        std::uint32_t __max_num_work_groups, std::uint32_t __max_block_size,
+                                        std::size_t __inputs_remaining)
+{
+    const std::uint32_t __num_sub_groups_local = __work_group_size / __sub_group_size;
+    const std::uint32_t __num_sub_groups_global = __num_sub_groups_local * __max_num_work_groups;
+    const std::uint32_t __max_inputs_per_subgroup = __max_block_size / __num_sub_groups_global;
+    const std::uint32_t __evenly_divided_remaining_inputs =
+        std::max(std::size_t{__sub_group_size},
+                 oneapi::dpl::__internal::__dpl_bit_ceil(__inputs_remaining) / __num_sub_groups_global);
+    const std::uint32_t __inputs_per_sub_group =
+        __inputs_remaining >= __max_block_size ? __max_inputs_per_subgroup : __evenly_divided_remaining_inputs;
+    const std::uint32_t __inputs_per_item = __inputs_per_sub_group / __sub_group_size;
+    return std::make_tuple(__num_sub_groups_local, __num_sub_groups_global, __inputs_per_sub_group, __inputs_per_item);
```

Comment:
Could you include `<tuple>` for `std::make_tuple`?

Reply:
Thanks, done.